state/graphql-sender/src/main/java/com/expediagroup/streamplatform/streamregistry/state/graphql/Credentials.java

```diff
@@ -0,0 +1,32 @@
+/**
+ * Copyright (C) 2018-2020 Expedia, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.expediagroup.streamplatform.streamregistry.state.graphql;
+
+import static kotlin.text.Charsets.UTF_8;
+
+import java.util.Base64;
+
+import lombok.RequiredArgsConstructor;
+
+@RequiredArgsConstructor
+public class Credentials {
+  private final String userName;
+  private final String password;
+
+  public String basic() {
+    return Base64.getEncoder().encodeToString((userName + ":" + password).getBytes(UTF_8));
```

Comment:
Apollo has a `Credentials` class that already does the encoding. Prefer not to have to maintain another class that does the same thing. Perhaps just have 2 parameters in `DefaultApolloClientFactory`?

Reply:
Except that it then makes it more complex to confirm that 'credentials' were provided

Reply:
I'm ok with there being a 'credentials' object as a container for name and pass, however I don't think this should be responsible for BASIC encoding. 

Nothing about name/pass credentials implies this encoding, thats decided on by the transport.

Reply:
Yea, that thought went through my head too, but abandoned it due to having to deal with 2 classes named `Credentials`. :D

But yes, a single class with non-nullable fields as a container would be preferable.

Reply:
done